app/addons/documents/views.js

```diff
@@ -564,15 +591,79 @@ function(app, FauxtonAPI, Components, Documents, Databases, Views, QueryOptions)
 
       this.toggleTrash();
       this.maybeHighlightAllButton();
+      this.updateExpandButtonLabel();
     },
 
     perPage: function () {
       return this.allDocsNumber.perPage();
+    },
+
+    // a bit dense, but necessary. It decides on what the Expand/Collapse button label should be and takes into
+    // account the default expand/collapse state, what rows are checked and their states
+    updateExpandButtonLabel: function () {
+
+      if (this.defaultDocsStateExpanded) {
+        this.setExpandButtonLabel(false);
+
+        // override the default label and show "Expand" if ALL checked rows have state === collapse (false)
+        if (this.numSelectedOnPage() && this.allCheckedRowsHaveState(false)) {
+          this.setExpandButtonLabel(true);
+        }
+      } else {
+        this.setExpandButtonLabel(true); // default show "Expand"
+
+        // show "Collapse" if ALL checked rows have state === expanded (true)
+        if (this.numSelectedOnPage() && this.allCheckedRowsHaveState(true)) {
+          this.setExpandButtonLabel(false);
+        }
+      }
+    },
+
+    setExpandButtonLabel: function(expanded) {
+      if (expanded) {
+        this.$('#collapse').html('<i class="icon-plus"></i> Expand');
+      } else {
+        this.$('#collapse').html('<i class="icon-minus"></i> Collapse');
+      }
+    },
+
+    setRowExpandedState: function (rows, state) {
+      _.each(rows, function(model) {
+        model.set('expanded', state);
+      });
+    },
+
+    allCheckedRowsHaveState: function (expanded) {
+      return _.every(this.getSelectedRows(), function (item) {
+        return item.get('expanded') === expanded;
+      });
+    },
+
+    // returns true if a subset of the rows on this page (1 to N-1) have been checked. That info is needed in
+    // determining the behaviour of the expand/collapse button
+    pageHasSelectedSubset: function () {
+      var numSelectedInPage = this.numSelectedOnPage();
+      return (numSelectedInPage !== 0 && numSelectedInPage !== this.collection.models.length);
+    },
+
+    // returns the models of selected items in the current page
+    getSelectedRows: function () {
+      if (!this.bulkDeleteDocsCollection) {
+        return [];
+      }
+      var allSelectedIDs = this.bulkDeleteDocsCollection.pluck('_id');
+      return _.filter(this.collection.models, function(item) {
+        return _.contains(allSelectedIDs, item.id);
+      }, this);
```

Comment:
I think we could use http://underscorejs.org/#some but maybe I am missing something